server/aws-lsp-partiql/src/server/tree-sitter-parser/parserUtil.ts

```diff
@@ -0,0 +1,34 @@
+import Parser from 'web-tree-sitter'
+import parserBase64 from '../tree-sitter-parser/tree-sitter-parser-inline'
+import wasmBinaryArray from '../tree-sitter-parser/tree-sitter-inline'
+
+// Initialize and prepare the parser
+export async function initParser() {
+    let wasmModuleUrl: string | null = null
+    if (typeof window !== 'undefined' && typeof window.URL.createObjectURL === 'function') {
+        // Browser
+        const wasmBlob = new Blob([wasmBinaryArray], { type: 'application/wasm' })
+        wasmModuleUrl = URL.createObjectURL(wasmBlob)
+    } else if (typeof process !== 'undefined' && typeof process.versions.node !== 'undefined') {
+        // Node.js
```

Comment:
Doing some thinking, as far as I can tell, the method using the blobs we use for browser and web worker should also work in Node. Why wouldn't we just use the same method for all paths to reduce these somewhat complex branching paths here?

It might lead to a slightly slower initialisation on node, but I think that's negligible and happens only once per server lifecycle.

Reply:
I understand the appeal of a unified approach using `Blob` across environments. However, while Node.js supports `Blob` objects, the `URL.createObjectURL()` function, which is primarily used in browser environments to create URLs pointing to in-memory files, is not natively supported in Node.js. 

I have attempted to use this method in Node.js result in errors because it cannot properly interpret or use the blob URLs created. That's why I've implemented a different handling for the Node.js environment.

![image](https://github.com/user-attachments/assets/0b300161-f09e-48fb-a1e3-aa1a7ba77199)

Reply:
Hmm weird, `URL.createObjectURL` does [exist](https://nodejs.org/api/url.html#urlcreateobjecturlblob) in Node >=16.7 but it indeed doesn't seem to work, not straight away at least.

Reply:
This should work across both Node, browser and webworker I think

```
// Initialize and prepare the parser
async function initParser() {
    await Parser.init({
        instantiateWasm(imports: any, succesCallback: any) {
            WebAssembly.instantiate(wasmBinaryArray, imports).then((arg) => succesCallback(arg.instance, arg.module));
        }
    })
    const parser = new Parser()
    const PartiQL = await Parser.Language.load(parserBase64)
    parser.setLanguage(PartiQL)
    return parser
}
```

I tested it works in Node, I'll also verify in webworker/browser.

Reply:
Also works in worker/browser